// vite-node/dist/source-map.mjs
import { install } from 'source-map-support';

let SOURCEMAPPING_URL = "sourceMa";
SOURCEMAPPING_URL += "ppingURL";
const VITE_NODE_SOURCEMAPPING_URL = `${SOURCEMAPPING_URL}=data:application/json;charset=utf-8;source=vite-node`;
const VITE_NODE_SOURCEMAPPING_REGEXP = new RegExp(`//# ${VITE_NODE_SOURCEMAPPING_URL};base64,(.+)`);
const OTHER_SOURCE_MAP_REGEXP = new RegExp(`//# ${SOURCEMAPPING_URL}=data:application/json[^,]+base64,(.+)`);
async function withInlineSourcemap(result) {
  const { code, map } = result;
  if (!map || code.includes(VITE_NODE_SOURCEMAPPING_URL))
    return result;
  if (OTHER_SOURCE_MAP_REGEXP.test(code))
    result.code = code.replace(OTHER_SOURCE_MAP_REGEXP, "");
  result.code = `${code}

//# ${VITE_NODE_SOURCEMAPPING_URL};base64,${Buffer.from(JSON.stringify(map), "utf-8").toString("base64")}
`;
  return result;
}
function extractSourceMap(code) {
  var _a;
  const mapString = (_a = code.match(VITE_NODE_SOURCEMAPPING_REGEXP)) == null ? void 0 : _a[1];
  if (mapString)
    return JSON.parse(Buffer.from(mapString, "base64").toString("utf-8"));
  return null;
}
function installSourcemapsSupport(options) {
  install({
    environment: "node",
    handleUncaughtExceptions: false,
    retrieveSourceMap(source) {
      const map = options.getSourceMap(source);
      if (map) {
        return {
          url: source,
          map
        };
      }
      return null;
    }
  });
}

export { extractSourceMap, installSourcemapsSupport, withInlineSourcemap };
